### PR TITLE
feat: add journal entry edit option

### PR DESCRIPTION
**Pull request by TeddyBear06**
Hi,

Before spending time on writing tests, I would like to get your opinion about this feature.

It is my first attempt to contribute to this project, so I probably need to adapt :-)

#### Before submitting the PR
- [X] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [X] If the PR is related to an issue or fix one, don't forget to indicate it. => https://github.com/monicahq/monica/issues/818
- [X] Indicate `[wip]` in the title of the PR it is is not final yet. Remove `[wip]` when ready. Otherwise the PR will be considered complete and rejected if it's not working.

### General checks
- [X] Make sure that the change you propose is the smallest possible.
- [X] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/index.md#conventional-commits) that the project follows.

### Front-end changes
- [X] If you change the UI, make sure to ask repositories administrators first about your changes by pinging @djaiss or @asbiin in this PR.
- [X] Screenshots are included if the PR changes the UI.
oLRvF62
db23p7U
- [ ] Front-end tests have been written with Cypress.

#### Backend/models changes
- [ ] The API has been updated.
- [ ] API's documentation has been added by submitting a pull request in the [marketing website repository](https://github.com/monicahq/marketing_site/pulls).
- [ ] Tests have been added for the new code.
- [ ] If you change a model, make sure the FakeContentTableSeeder is updated. We need seeders to develop locally and generate fake data.

#### If the code changes the SQL schema
- [ ] Make sure exporting account data as SQL is still working.
- [ ] Make sure your changes do not break importing data with `vCard` and `.csv` files.
- [ ] Make sure account reset and deletion still work.

#### Other tasks
- [ ] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.
- [ ] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.
- [ ] If it's relevant and worth mentioning, create a changelog entry for this change. The changelog entry will appear inside the UI for all users to see. To know if your change is worth the creation of a changelog entry, [read the documentation](https://github.com/monicahq/monica/blob/master/docs/administrators/tips.md#when-is-it-relevant-to-create-a-changelog-entry).
- [ ] Don't forget to [ask for a free account](mailto:regis@monicahq.com) on https://monicahq.com as anyone who contributes can request a free account.

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -196,6 +196,8 @@
 
         Route::get('/journal/add', 'JournalController@create')->name('create');
         Route::post('/journal/create', 'JournalController@save')->name('save');
+        Route::get('/journal/entries/{entry}/edit', 'JournalController@edit')->name('edit');
+        Route::put('/journal/{entryId}', 'JournalController@update')->name('update');
```
Comment (asbiin) [inline routes/web.php]: It should be
```suggestion
        Route::put('/journal/entries/{entryId}', 'JournalController@update')->name('update');
```
to be more compliant

File: app/Models/Journal/JournalEntry.php [hunk 2]
```diff
@@ -65,6 +65,26 @@ public function add($resourceToLog)
...
     }
 
+    /**
+     * Update an entry in the journal.
+     * @param mixed $resourceToLog
+     */
+    public function edit($resourceToLog)
+    {
+        
+        $journalEntry = self::where(
```
[the remark this answers is not in the document]
Comment (TeddyBear06) [reply app/Models/Journal/JournalEntry.php]: Got it :-)

**Comment by TeddyBear06** [comment]
Updated :-)

Let me know if you want anything else.

Kind regards,

**Comment by asbiin** [comment]
@TeddyBear06 thanks a lot! As you can see I've updated some of your code, for the better.

**Comment by TeddyBear06** [comment]
@asbiin Wonderful! Thanks a lot! I think this little feature will be great :-)

**Comment by danielsundermeier** [comment]
i would suggest a little change in /Users/daniel/code/monicahq/monica/resources/assets/js/components/journal/partials/JournalContentEntry.vue:38
- :href="'/journal/entries/' + entry.id + '/edit'"
+ :href="'journal/entries/' + entry.id + '/edit'"

right now it does not work if monica is not installed in the root directory

(i don't know how to do it myself)

**Comment by asbiin** [comment]
@danielsundermeier thanks for the report ! it's fixed.

**Comment by TeddyBear06** [comment]
@danielsundermeier Good catch! @asbiin Thanks!

**Pull request merged**